bot/cogs/filtering.py

```diff
@@ -126,6 +136,54 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    @staticmethod
+    def get_name_matches(name: str) -> List[re.Match]:
+        """Check bad words from passed string (name). Return list of matches."""
+        matches = []
+        for pattern in WATCHLIST_PATTERNS:
+            match = pattern.search(name)
+            if match:
+                matches.append(match)
```

Comment:
Seems like a good use case for a walrus (assuming the linter supports them):
```suggestion
            if match := pattern.search(name):
                matches.append(match)
```